sanitize port name before converting the topology

```
2024-11-15 08:41:17,002 - ERROR [kytos.napps.kytos/sdx] [main.py:388:post_topology_to_sdxlc] (AnyIO worker thread) Failed to send topoloty to SDX-LC: {
  "detail": "'MON FOR-LAN-SW02 et-0/0/2 -- F' does not match '^[A-Za-z0-9_.,/-]*$' - 'nodes.0.ports.0.name'",
  "status": 400,
  "title": "Bad Request",
  "type": "about:blank"
}
 - Traceback: Traceback (most recent call last):,   File "//var/lib/kytos/napps/kytos/sdx/main.py", line 384, in post_topology_to_sdxlc,     assert response.status_code == 200, response.text, AssertionError: {,   "detail": "'MON FOR-LAN-SW02 et-0/0/2 -- F' does not match '^[A-Za-z0-9_.,/-]*$' - 'nodes.0.ports.0.name'",,   "status": 400,,   "title": "Bad Request",,   "type": "about:blank", }, ,
```

--- a/convert_topology.py
+++ b/convert_topology.py
@@ -134,8 +134,12 @@ class ParseConvertTopology:
 
         sdx_port = {}
         sdx_port["id"] = self.get_port_urn(interface)
-        sdx_port["name"] = interface["metadata"].get("port_name", "")[:30]
-        if not sdx_port["name"]:
+        if interface["metadata"].get("port_name"):
+            port_name = interface["metadata"]["port_name"]
+            port_name = re.sub(r"\s+", "_", port_name)
+            port_name = re.sub("[^A-Za-z0-9_.,/-]", "", port_name)
+            sdx_port["name"] = port_name[:30]
+        else:
             sdx_port["name"] = interface["name"][:30]
         sdx_port["node"] = f"urn:sdx:node:{self.oxp_url}:{sdx_node_name}"
         sdx_port["type"] = self.get_type_port_speed(str(interface["speed"]))

